### DISCUSSION TITLE
Exporting to DXF with fixed size

### DISCUSSION THREAD

**Discussion by matejkac** [microsoft/maker.js#576] [Q&A] [answered]
I would like to export DXF file in a fixed size.

I am creating curved text around a ring and would like to export just the text but text will not always fill entire ring.
I need the DXF file to be of the size of entire ring but when I export it, it only exports the part where text is in. 

For example, if radius of ring is 60 mm, then I need the DXF dimensions to be 120x120 mm, with the center point being the center of the ring.

Are there any parameters to set for exporting?

**Comment by danmarshall** [comment]
You can set the `origin` of your model to position it.

**Reply by matejkac** [reply]
Thanks for the answer. This works for other model types, not for text models. Is there any other way to position it?

**Reply by danmarshall** [reply]
can you "wrap" your model(s) in a parent model, then set the origin on the parent? If that doesn't work, maybe you can give some example code.

**Reply by matejkac** [reply]
That works, the issue is however when sending the generated DXF to engraving software, it will engrave the enclosing model also.

**Reply by danmarshall** [reply]
Sorry, I meant to say an "empty" model - that doesn't have its own paths. It wouldn't be a container visually, just an object that parents yours.

**Reply by matejkac** [reply]
Is this correct way of making an empty model?
`    const empty_model = {
        origin: [-60,-70],
        models: { topTextModel, bottomTextModel } }
`

**Reply by danmarshall** [reply]
yes exactly.

**Reply by danmarshall** [reply]
You may also want to take a look at [originating](https://maker.js.org/docs/intermediate-drawing/#Originating).

**Reply by matejkac** [reply]
Still the same result. If I look at the generated DXF , it doesnt include EXTMIN and EXTMAX which tell the limits.

**Reply by danmarshall** [reply]
True, EXTMIN and EXTMAX aren't in the output. But entities are placed in proper coordinate space. Consider this code, where c1 and c2 are origins.

```js
  const c1 = [0, 0];
  const c2 = [10, 10];
  
  this.paths = {
    example: new makerjs.paths.Circle(c1, 10)
  };
 ```

using c1: 
<img width="192" alt="image" src="https://github.com/microsoft/maker.js/assets/11507384/d95af994-6a84-472b-8785-f5c72b34127e">

using c2: 
<img width="202" alt="image" src="https://github.com/microsoft/maker.js/assets/11507384/898207aa-a7ff-4f01-993e-fa282cef6211">

Does your software have an option to keep things in their exact location?

**Reply by matejkac** [reply]
I have found a workaround. If I use ` const rect_text1 = new makerjs.models.Rectangle(text1);` i can then find the origin point of the enclosing rectangle. I can then export each text in its own DXF file and position it accordingly in the engraving software. 
I would still prefer a way to export as a single file.

**Reply by danmarshall** [reply]
can you use the `rect_text1` dimensions and programmatically append the EXTMIN/MAX to the dxf string?

**Reply by matejkac** [reply, marked as answer]
And if we combine our ideas, I can use the surrounding rectangle to surround the empty and getting the origin point I can position with just a single file. This work, thanks for your help. Great library.

**Comment by matejkac** [comment]
So the software works by setting the justification of each element in the template file. When you import the DXF, it aligns the element by drawing a rectangle according to the justification setting.

**Comment by matejkac** [comment]
> can you use the `rect_text1` dimensions and programmatically append the EXTMIN/MAX to the dxf string?
I will try that also, thanks for the tip. For now it works by inserting the generated DXF file and then sending the command to move it to the proper place.
